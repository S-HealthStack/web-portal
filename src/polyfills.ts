import smoothScroll from 'smoothscroll-polyfill';
import _isNull from 'lodash/isNull';
import _isObject from 'lodash/isObject';
import browser from 'src/common/utils/browser';

type ArgumentTypes<F extends () => unknown> = F extends (...args: infer A) => unknown ? A : never;

// safari smooth scroll fix
if (browser.isSafari) {
  smoothScroll.polyfill();

  const prevScrollTo = window.scrollTo;

  // issue: https://github.com/iamdustan/smoothscroll/issues/161
  Object.assign(window, {
    scrollTo: ((...args) => {
      if (_isObject(args[0])) {
        args[0].left = _isNull(args[0].left) ? undefined : args[0].left;
        args[0].top = _isNull(args[0].top) ? undefined : args[0].top;
      } else {
        args[0] = _isNull(args[0]) ? undefined : args[0];
        args[1] = _isNull(args[1]) ? undefined : args[1];
      }

      prevScrollTo(...(args as ArgumentTypes<typeof window.scrollTo>));
    }) as typeof window.scrollTo,
  });
}
