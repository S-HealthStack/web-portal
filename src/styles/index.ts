export { default as theme } from './theme';
export {
  colorsFn as colors,
  typographyFn as typography,
  boxShadowFn as boxShadow,
  animationFn as animation,
} from './themeHelpers';
export { default as GlobalStyles } from './GlobalStyles';
export { px } from './utils';
