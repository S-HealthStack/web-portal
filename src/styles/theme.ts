import { css, DefaultTheme } from 'styled-components';
import { px } from './utils';

const colors = {
  // gray gradations
  black: '#000000',
  black40: 'rgba(0, 0, 0, 0.4)',
  black08: 'rgba(0, 0, 0, 0.08)',

  // light mode
  surface: '#FFFFFF',
  background: '#F7F8FA',
  onPrimary: '#FFFFFF',
  backgroundOnPrimary: '#FFFFFF',
  backgroundSurface: '#FFFFFF',
  backgroundOnSurface: '#2D2D2D',
  primary: '#4475E3',
  primaryLight: '#ECF1FC',
  primaryHovered: '#2F65E0',
  primary05: '#F6F8FE',
  primary10: '#ECF1FC',
  primary30: '#C7D6F7',
  primaryDisabled: '#B3C6F1',
  primaryLightPressed: '#E3EAFB',
  primaryLightFocused: '#C7D6F7',
  textPrimary: '#474747',
  primaryWhite: '#FFFFFF',
  disabled: '#DBDBDB',
  disabled20: 'rgba(219, 219, 219, 0.2)',
  onDisabled: '#A1A1A1',

  textPrimaryBlue: '#4475E3',
  textPrimaryDark: '#474747',
  textSecondaryGray: '#808080',
  textDisabled: '#838383',
  onSurface: '#2D2D2D',
  secondarySkyblue: '#00B0D7', // replace
  backgoundLight: '#F0F0F0',
  primaryBlueHovered: '#2F65E0',
  primaryBluePressed: '#2057D5',

  statusSuccess: '#2DC008',
  statusSuccess10: '#DCF2EA',
  statusSuccessText: '#00825D',

  statusError: '#FF3333',
  statusError10: '#FFEBEB',
  statusErrorText: '#D14343',

  statusWarning: '#F5BF00',
  statusWarning10: '#FFF7D0',
  statusWarningText: '#A36400',

  secondaryViolet: '#944ED7',
  secondaryVioletHovered: '#6A3799',
  secondaryViolet10: '#F4EDFB',

  secondaryTangerine: '#F39C18',
  secondaryTangerineHovered: '#CC8314',
  secondaryTangerine10: '#FEF5E8',

  secondarySkyBlue: '#00B0D7',
  secondarySkyBlueHovered: '#0092B3',
  secondarySkyBlue10: '#E5F7FB',

  secondaryGreen: '#1AD598',
  secondaryGreenHovered: '#16B37F',
  secondaryGreen10: '#E8FBF5',

  secondaryRed: '#FA5F4F',
  secondaryRedHovered: '#CC4E40',
  secondaryRed10: '#FFEFED',
};

export type SpecColorType = keyof typeof colors;

const weightToCssValue = {
  regular: 400,
  upperregular: 450,
  medium: 500,
  semibold: 600,
  bold: 700,
};

const fontSpec = (
  size: number,
  lineHeight: number | string,
  weight: keyof typeof weightToCssValue,
  fontFamily: string
) => css`
  font-family: ${fontFamily}, sans-serif;
  font-size: ${px(size)};
  line-height: ${typeof lineHeight === 'string' ? lineHeight : px(lineHeight)};
  font-weight: ${weightToCssValue[weight]};
  font-style: normal;
`;

const typography = {
  // Inter typeface
  labelRegular: fontSpec(10, 15, 'regular', 'Inter'),
  labelSemibold: fontSpec(10, 15, 'semibold', 'Inter'),

  bodyXSmallRegular: fontSpec(12, 18, 'regular', 'Inter'),
  bodyXSmallSemibold: fontSpec(12, 18, 'semibold', 'Inter'),
  bodySmallRegular: fontSpec(14, 21, 'regular', 'Inter'),
  bodySmallSemibold: fontSpec(14, 21, 'semibold', 'Inter'),
  bodyMediumRegular: fontSpec(16, 24, 'regular', 'Inter'),
  bodyMediumSemibold: fontSpec(16, 24, 'semibold', 'Inter'),
  bodyLargeRegular: fontSpec(18, 26, 'regular', 'Inter'),
  bodyLargeSemibold: fontSpec(18, 26, 'semibold', 'Inter'),

  headingXSmall: fontSpec(16, 21, 'semibold', 'Inter'),
  headingSmall: fontSpec(18, 23, 'semibold', 'Inter'),
  headingXMedium: fontSpec(18, 27, 'semibold', 'Inter'),
  headingXMediumRegular: fontSpec(18, 27, 'regular', 'Inter'),
  headingMedium: fontSpec(24, 32.2, 'semibold', 'Inter'),
  headingLargeSemibold: fontSpec(32, 42, 'semibold', 'Inter'),
  headingXLargeSemibold: fontSpec(48, 58, 'semibold', 'Inter'),
  headingXXLargeSemibold: fontSpec(60, 72, 'semibold', 'Inter'),

  // Fira Code typeface
  query14: fontSpec(14, '130%', 'upperregular', 'Fira Code'),
  query12: fontSpec(12, '150%', 'upperregular', 'Fira Code'),
};

const boxShadow = {
  card: `0 0 ${px(2)} ${px(0)} rgba(0,0,0,0.15)`,
  avatar: `0 ${px(8)} ${px(10)} ${px(2)} 0 rgba(0,0,0,0.12)`,
  studyAvatar: `0 ${px(15)} ${px(20)} rgba(180,180,180,0.3)`,
  previewScreen: `0 ${px(2)} ${px(4)} rgba(71,71,71,0.25)`,
};

const animation = {
  defaultTiming: 'cubic-bezier(0.45, 0.05, 0, 1)',
  rippleTiming: 'cubic-bezier(0.4, 0, 0.2, 1)',
};

declare module 'styled-components' {
  export interface DefaultTheme {
    colors: typeof colors;
    typography: typeof typography;
    boxShadow: typeof boxShadow;
    animation: typeof animation;
  }
}

const theme: DefaultTheme = {
  colors,
  typography,
  boxShadow,
  animation,
};

export default theme;
