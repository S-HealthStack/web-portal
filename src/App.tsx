import React from 'react';
import { Provider } from 'react-redux';
import { Route, Switch } from 'react-router-dom';
import { ConnectedRouter } from 'connected-react-router';
import { ThemeProvider } from 'styled-components';
import { DndProvider } from 'react-dnd';
import { TouchBackend } from 'react-dnd-touch-backend';

import { store } from 'src/modules/store/store';
import { history, Path } from 'src/modules/navigation/store';
import PrivateRoute from 'src/modules/navigation/private-route/PrivateRoute';
import MainLayout from 'src/modules/main-layout/MainLayout';
import SignInScreen from 'src/modules/auth/signin/SignInScreen';
import AccountActivationScreen from 'src/modules/auth/signin/AccountActivationScreen';
import CreateStudyScreen from 'src/modules/auth/signin/CreateStudyScreen';
import { TooltipProvider, TooltipsList } from 'src/common/components/Tooltip';
import { theme, GlobalStyles } from 'src/styles';

import 'src/modules/auth/authProvider';

const App = () => (
  <ThemeProvider theme={theme}>
    <TooltipProvider>
      <DndProvider backend={TouchBackend} options={{ enableMouseEvents: true }}>
        <Provider store={store}>
          <GlobalStyles />
          <ConnectedRouter history={history}>
            <Switch>
              <Route path={Path.SignIn} component={SignInScreen} />
              <Route path={Path.AccountActivation} component={AccountActivationScreen} />
              <PrivateRoute path={Path.CreateStudy}>
                <CreateStudyScreen />
              </PrivateRoute>
              <PrivateRoute>
                <MainLayout />
              </PrivateRoute>
            </Switch>
          </ConnectedRouter>
        </Provider>
        <TooltipsList />
      </DndProvider>
    </TooltipProvider>
  </ThemeProvider>
);

export default App;
