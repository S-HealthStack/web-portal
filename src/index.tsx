import React from 'react';
import { createRoot } from 'react-dom/client';

import printAppVersion from 'src/common/utils/printAppVersion';

import './polyfills';
import App from './App';

printAppVersion();

const container = document.getElementById('root');
const root = createRoot(container as HTMLElement);

root.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>
);
