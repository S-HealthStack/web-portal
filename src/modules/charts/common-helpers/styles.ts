import styled, { css } from 'styled-components';

import { colors, typography, px, animation } from 'src/styles';
import { LINE_CLASS_NAME } from '../common-components/Line';
import { AREA_CLASS_NAME } from '../common-components/Area';
import { DOT_CLASS_NAME } from '../common-components/Dot';

interface ContainerProps {
  width: number;
  height: number;
}

export const DOTS_CONTAINER_CLASS_NAME = 'dots';
export const X_AXIS_CLASS_NAME = 'xAxis';
export const Y_AXIS_CLASS_NAME = 'yAxis';
export const ZOOM_CLASS_NAME = 'zoom';

export const Container = styled.div.attrs<ContainerProps>(({ width, height }) => ({
  style: {
    width: px(width),
    height: px(height),
    position: 'relative',
  },
}))<ContainerProps>``;

export const TooltipContent = styled.div`
  display: flex;
  align-items: center;
`;

export const TooltipNameBlock = styled.div`
  display: flex;
  margin-right: ${px(8)};

  p {
    ${typography.labelSemibold};
    color: ${colors.surface};
    margin: 0;
  }
`;

export const TooltipDataBlock = styled.div`
  display: flex;
  flex-direction: column;

  p {
    ${typography.labelRegular};
    color: ${colors.surface};
    margin: 0;
  }
`;

export const TooltipColorPoint = styled.div<{ color: string }>`
  width: ${px(12)};
  height: ${px(12)};
  background-color: ${({ color }) => color};
  border-radius: 50%;
  margin-right: ${px(8)};
`;

export const brushOverlayStyles = css`
  fill: ${colors.primary};
  fill-opacity: 0.1;
  pointer-events: none;
`;

export const brushSelectionStyles = css`
  fill: ${colors.primary};
  fill-opacity: 1;
`;

export const brushHandleCommonStyles = css`
  stroke-width: ${px(1)};
  stroke: ${colors.primary};
  stroke-opacity: 0.3;
  fill: ${colors.onPrimary};
`;

export const StyledSvg = styled.svg<{
  focusWidth: number;
  $contextVisible: boolean;
}>`
  .${X_AXIS_CLASS_NAME} {
    .tick line {
      opacity: 0;
    }

    .tick {
      :first-child text {
        text-anchor: start;
      }
      :last-child text {
        text-anchor: end;
      }
    }
  }

  .${Y_AXIS_CLASS_NAME} {
    .tick line {
      stroke: ${colors.updBackgoundLight};
    }
  }

  .context {
    opacity: ${({ $contextVisible }) => ($contextVisible ? 1 : 0)};
    transition: opacity 0.3s ${animation.defaultTiming};
    .brushX {
      .overlay {
        ${brushOverlayStyles};
        height: ${px(4)};
      }
      .selection {
        ${brushSelectionStyles};
        height: ${px(22)};
        fill: url(#horizontalGradient);
      }
      .handle {
        ${brushHandleCommonStyles};
        height: ${px(22)};
        width: ${px(6)};
      }
    }
    .brushY {
      .overlay {
        ${brushOverlayStyles};
        width: ${px(4)};
      }
      .selection {
        ${brushSelectionStyles};
        width: ${px(22)};
        fill: url(#verticalGradient);
      }
      .handle {
        ${brushHandleCommonStyles};
        height: ${px(6)};
        width: ${px(22)};
      }
    }
  }

  .focus {
    cursor: move;
  }

  .${LINE_CLASS_NAME} {
    clip-path: url(#clip);
    transition: stroke-opacity 0.3s ${animation.defaultTiming};
  }

  .${AREA_CLASS_NAME} {
    clip-path: url(#clip);
    transition: fill-opacity 0.3s ${animation.defaultTiming};
  }

  .${DOTS_CONTAINER_CLASS_NAME} {
    clip-path: url(#clip);
    .${DOT_CLASS_NAME} {
      transition: fill-opacity 0.3s ${animation.defaultTiming};
    }
  }

  .zoom {
    fill: none;
    pointer-events: all;
  }
`;
