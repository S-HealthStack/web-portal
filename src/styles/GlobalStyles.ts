import { createGlobalStyle } from 'styled-components';
import { px } from 'src/styles/utils';

export const previewDropdownMenuClassName = 'preview-dropdown-menu';
export const PREVIEW_SCALE = 0.75;

export default createGlobalStyle`
  * {
    box-sizing: border-box;
  }

  html {
    font-size: 100%;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
  }

  html, body {
    height: 100%;
    width: 100%;
    margin: 0;
    position: fixed;
    overflow: hidden;
    overscroll-behavior-y: none;
    background: ${({ theme }) => theme.colors.background};
    user-select: none;
    -webkit-user-drag: none;
    font-family: Inter, sans-serif;
  }
  
  input,
  input:focus {
    ::-webkit-textfield-decoration-container {
      visibility: hidden;
    }
  }

  #root {
    height: 100%;
    width: 100%;
    display: flex;
    flex-direction: column;
    overflow: hidden;
  }
  
  *:focus,
  *:focus-visible {
    outline: none;
  }
  
  .recharts-surface {
    overflow: visible;
  }

  .${previewDropdownMenuClassName} {
    border: none;
    filter: drop-shadow(
      0 
      ${px(5 * PREVIEW_SCALE)} 
      ${px(12 * PREVIEW_SCALE)} 
      rgba(0, 0, 0, 0.12)
    ); // TODO unknown color
    margin-top: ${px(8 * PREVIEW_SCALE)};
  }
`;
