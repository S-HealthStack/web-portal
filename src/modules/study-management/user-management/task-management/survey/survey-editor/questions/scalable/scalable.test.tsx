import React from 'react';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components/';
import { TouchBackend } from 'react-dnd-touch-backend';
import { DndProvider } from 'react-dnd';
import { theme } from 'src/styles';
import handler from './index';

/* eslint-disable @typescript-eslint/no-explicit-any */

describe('scalable', () => {
  it('type should be slider', async () => {
    expect(handler.type).toEqual('slider');
  });

  it('should return empty question', async () => {
    expect(handler.createEmpty()).toEqual({
      type: 'slider',
      id: 'survey_question1',
      title: '',
      description: '',
      answers: [
        { id: 'survey_question2', label: '', value: 0 },
        { id: 'survey_question3', label: '', value: 10 },
      ],
      options: {
        optional: false,
      },
    });
  });

  it('isEmpty should return true', async () => {
    expect(
      handler.isEmpty({
        id: '1',
        type: 'slider',
        title: '',
        description: '',
        answers: [
          { id: 'survey_question2', label: '', value: 0 },
          { id: 'survey_question3', label: '', value: 10 },
        ],
        options: {
          optional: false,
        },
      })
    ).toBeTrue();
  });

  const notEmptyQuestionItem = {
    id: '1',
    type: 'single',
    title: 'Title',
    description: 'Description',
    answers: [
      { id: 'survey_question2', label: 'Label1', value: 0 },
      { id: 'survey_question3', label: 'Label2', value: 10 },
    ],
    options: {
      optional: false,
    },
  };

  it('isEmpty should return false', async () => {
    expect(handler.isEmpty(notEmptyQuestionItem as any)).toBeFalse();
  });

  it('[NEGATIVE] isEmpty should not crash if invalid data received', async () => {
    expect(handler.isEmpty({} as any)).toBeUndefined();
  });

  it('convertFromOtherType should return converted question', async () => {
    const qi = {
      type: 'multiple',
      id: 'Question1',
      title: 'Title',
      description: 'Description',
      skipLogic: undefined,
      options: {
        optional: false,
        includeOther: false,
      },
      answers: [],
    };
    expect(handler.convertFromOtherType(qi as any)).toEqual({
      ...qi,
      type: 'slider',
      answers: [
        { id: 'survey_question2', label: '', value: 0 },
        { id: 'survey_question3', label: '', value: 10 },
      ],
      options: {
        optional: false,
      },
    });
  });

  it('[NEGATIVE] convertFromOtherType should return empty question', async () => {
    expect(handler.convertFromOtherType({} as any)).toEqual({
      type: 'slider',
      id: undefined,
      title: undefined,
      description: undefined,
      answers: [
        { id: 'survey_question2', label: '', value: 0 },
        { id: 'survey_question3', label: '', value: 10 },
      ],
      options: {
        optional: false,
      },
      sectionId: undefined,
      skipLogic: undefined,
    });
  });

  const taskItemFromApi = {
    name: 'name',
    sequence: 1,
    type: 'QUESTION',
    contents: {
      title: 'Question Title',
      explanation: 'Question Description',
      required: false,
      type: 'SCALE',
      properties: {
        tag: 'SLIDER',
        lowLabel: 'Low Label',
        highLabel: 'High Label',
        low: 1,
        high: 9,
      },
    },
  };

  it('fromApi should return converted question', async () => {
    expect(handler.fromApi(taskItemFromApi as any)).toEqual({
      type: 'slider',
      id: 'name',
      title: 'Question Title',
      description: 'Question Description',
      answers: [
        { id: 'low', label: 'Low Label', value: 1 },
        { id: 'high', label: 'High Label', value: 9 },
      ],
      options: {
        optional: true,
      },
    });
  });

  it('[NEGATIVE] fromApi should return undefined if task type is not a question', async () => {
    expect(handler.fromApi({ ...taskItemFromApi, type: 'ROW' })).toBeUndefined();
  });

  it('[NEGATIVE] fromApi should return undefined if question type is not a scale', async () => {
    expect(
      handler.fromApi({
        ...taskItemFromApi,
        contents: { ...taskItemFromApi.contents, type: 'TEXT' },
      } as any)
    ).toBeUndefined();
  });

  it('toApi should return converted question', async () => {
    expect(handler.toApi(notEmptyQuestionItem as any)).toEqual({
      name: '1',
      type: 'QUESTION',
      sequence: 0,
      contents: {
        title: 'Title',
        explanation: 'Description',
        required: true,
        type: 'SCALE',
        properties: {
          tag: 'SLIDER',
          low: 0,
          lowLabel: 'Label1',
          high: 10,
          highLabel: 'Label2',
        },
      },
    });
  });

  it('[NEGATIVE] toApi should not crash with empty data', async () => {
    expect(handler.toApi({} as any)).toBeUndefined();
  });

  it("[NEGATIVE] transformAnswersOnQuestionChange should return empty object if types don't match", async () => {
    expect(
      handler.transformAnswersOnQuestionChange({
        question: notEmptyQuestionItem as any,
        previousQuestion: { type: 'dropdown' } as any,
        answers: { 1: '1' },
      })
    ).toEqual({});
  });

  it('[NEGATIVE] transformAnswersOnQuestionChange should return empty object if invalid answers data', async () => {
    expect(
      handler.transformAnswersOnQuestionChange({
        question: notEmptyQuestionItem as any,
        previousQuestion: notEmptyQuestionItem as any,
        answers: { value: '0' },
      })
    ).toEqual({});
  });

  it('transformAnswersOnQuestionChange should return value', async () => {
    expect(
      handler.transformAnswersOnQuestionChange({
        question: notEmptyQuestionItem as any,
        previousQuestion: notEmptyQuestionItem as any,
        answers: { value: 5 },
      })
    ).toEqual({ value: 5 });
  });

  it('renderEditorContent should render editor content', async () => {
    const onChange = jest.fn();
    const confirmOptionRemoval = jest.fn();

    const { baseElement } = render(
      <ThemeProvider theme={theme}>
        <DndProvider backend={TouchBackend} options={{ enableMouseEvents: true }}>
          {handler.renderEditorContent({
            surveyId: '',
            question: notEmptyQuestionItem as any,
            onChange,
            confirmOptionRemoval,
          })}
        </DndProvider>
      </ThemeProvider>
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('[NEGATIVE] renderEditorContent should return null if question is undefined', async () => {
    const onChange = jest.fn();
    const confirmOptionRemoval = jest.fn();

    expect(
      handler.renderEditorContent({
        surveyId: '',
        question: undefined as any,
        onChange,
        confirmOptionRemoval,
      })
    ).toBeNull();
  });

  it('renderPreviewContent should render preview content', async () => {
    const onAnswersChange = jest.fn();
    const onChange = jest.fn();

    const { baseElement } = render(
      <ThemeProvider theme={theme}>
        {handler.renderPreviewContent({
          question: notEmptyQuestionItem as any,
          answers: { value: 0 },
          onAnswersChange,
          onChange,
        })}
      </ThemeProvider>
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('[NEGATIVE] renderPreviewContent should return null if maxAnswer is not defined', async () => {
    const onAnswersChange = jest.fn();
    const onChange = jest.fn();

    expect(
      handler.renderPreviewContent({
        question: { ...notEmptyQuestionItem, answers: [{ id: '1', label: '1', value: 0 }] } as any,
        answers: { value: 0 },
        onChange,
        onAnswersChange,
      })
    ).toBeNull();
  });
});
