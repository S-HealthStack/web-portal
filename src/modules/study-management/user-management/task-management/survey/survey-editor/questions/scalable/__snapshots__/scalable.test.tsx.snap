// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scalable renderEditorContent should render editor content 1`] = `
.c5 {
  position: relative;
  width: 100%;
}

.c8 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #474747;
  grid-area: label;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.c9 {
  grid-area: right-icon;
}

.c6 {
  width: 100%;
  height: 3.5rem;
  padding: 0 1rem;
  padding-right: 0.5rem;
  box-sizing: border-box;
  display: grid;
  grid-template-areas: ' label right-icon';
  grid-template-columns: 1fr 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.25rem;
  background-color: #F7F8FA;
  border: 0.0625rem solid transparent;
  border-top-width: 0;
  -webkit-transition: all 300ms cubic-bezier(0.45,0.05,0,1);
  transition: all 300ms cubic-bezier(0.45,0.05,0,1);
}

.c6:hover {
  cursor: pointer;
}

.c7 {
  border-top-width: 0.0625rem;
  border-color: !important;
  border-radius: 0.25rem;
}

.c7:hover {
  border-color: #2F65E0;
}

.c10 {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 400;
  font-style: normal;
  color: #474747;
  box-sizing: border-box;
  height: 3.5rem;
  width: 100%;
  margin: 0;
  padding: 1rem;
  background-color: #F7F8FA;
  border: 0.0625rem solid #F7F8FA;
  border-radius: 0.25rem;
  -webkit-transition: border 300ms cubic-bezier(0.45,0.05,0,1);
  transition: border 300ms cubic-bezier(0.45,0.05,0,1);
  caret-color: #4475E3;
}

.c10:hover:enabled {
  border-color: #2F65E0;
}

.c10:active:enabled,
.c10:focus-visible {
  outline: none;
  border-color: #4475E3;
}

.c10:disabled {
  color: rgba(0,0,0,0.38);
  border-color: transparent;
  background-color: #f8f8f8;
}

.c10::-webkit-input-placeholder {
  color: #808080;
}

.c10::-moz-placeholder {
  color: #808080;
}

.c10:-ms-input-placeholder {
  color: #808080;
}

.c10::placeholder {
  color: #808080;
}

.c10:disabled::-webkit-input-placeholder {
  color: #DBDBDB;
}

.c10:disabled::-moz-placeholder {
  color: #DBDBDB;
}

.c10:disabled:-ms-input-placeholder {
  color: #DBDBDB;
}

.c10:disabled::placeholder {
  color: #DBDBDB;
}

.c10:-webkit-autofill,
.c10:-webkit-autofill:hover,
.c10:-webkit-autofill:focus {
  border: 0.0625rem solid #4475E3;
  -webkit-transition: background-color 5000s ease-in-out 0s;
  transition: background-color 5000s ease-in-out 0s;
  background-color: #F7F8FA !important;
  background-image: unset !important;
  -webkit-text-fill-color: #838383;
}

.c10:-webkit-autofill:focus {
  border-color: #4475E3;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1rem;
  padding: 0.625rem 0rem 0.5rem;
}

.c3 {
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 2rem;
}

.c1 > .c2:first-child {
  grid-template-columns: 7.5rem 5.75rem;
}

.c1 > .c2:last-child {
  grid-template-columns: 2.25rem 14.125rem;
}

.c4 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #474747;
}

@media screen and (min-width:64rem) {
  .c0 {
    padding-left: 0.5rem;
    margin-bottom: 0rem;
  }
}

@media screen and (min-width:90rem) {
  .c0 {
    padding: 0.625rem 0rem 0.5rem;
  }
}

@media screen and (min-width:64rem) {
  .c1 {
    gap: 2rem;
  }
}

@media screen and (min-width:90rem) {
  .c1 {
    gap: 2rem;
  }
}

@media screen and (min-width:64rem) {
  .c1 > .c2:first-child {
    grid-template-columns: 7.5rem 5.75rem;
  }
}

@media screen and (min-width:90rem) {
  .c1 > .c2:first-child {
    grid-template-columns: 7.5rem 5.75rem;
  }
}

<body>
  <div>
    <div
      class="c0"
    >
      <div
        class="c1"
      >
        <div
          class="c2 c3"
        >
          <div
            class="c4"
          >
            Minimum number
          </div>
          <div
            class="c5"
          >
            <div
              class="c6 c7"
              data-testid="value-item"
            >
              <div
                class="c8"
              >
                0
              </div>
              <svg
                class="c9"
              />
            </div>
          </div>
        </div>
        <div
          class="c2 c3"
        >
          <div
            class="c4"
          >
            Label
          </div>
          <input
            class="c10"
            maxlength="20"
            value="Label1"
          />
        </div>
      </div>
      <div
        class="c1"
      >
        <div
          class="c2 c3"
        >
          <div
            class="c4"
          >
            Maximum number
          </div>
          <div
            class="c5"
          >
            <div
              class="c6 c7"
              data-testid="value-item"
            >
              <div
                class="c8"
              >
                10
              </div>
              <svg
                class="c9"
              />
            </div>
          </div>
        </div>
        <div
          class="c2 c3"
        >
          <div
            class="c4"
          >
            Label
          </div>
          <input
            class="c10"
            maxlength="20"
            value="Label2"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`scalable renderPreviewContent should render preview content 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  position: relative;
  height: 4.375rem;
}

.c7 {
  width: 100%;
  background-color: #ECF1FC;
  height: 0.125rem;
}

.c2 {
  width: 100%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding: 0 0.375rem;
}

.c3 {
  cursor: pointer;
  height: 2.25rem;
  position: relative;
  top: -0.3125rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  width: calc((100% / 10) / 2);
}

.c5 {
  cursor: pointer;
  height: 2.25rem;
  position: relative;
  top: -0.3125rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: calc(100% / 10);
}

.c6 {
  cursor: pointer;
  height: 2.25rem;
  position: relative;
  top: -0.3125rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  width: calc((100% / 10) / 2);
}

.c4 {
  cursor: pointer;
  width: 0.125rem;
  position: relative;
  top: 1.25rem;
  height: 0.375rem;
  background-color: #4475E3;
}

.c9 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  font-family: Inter,sans-serif;
  font-size: 0.625rem;
  line-height: 0.9375rem;
  font-weight: 400;
  font-style: normal;
  line-height: 0.8125rem;
  color: #474747;
  position: absolute;
  top: 1.75rem;
  left: 0;
}

.c10 {
  color: #808080;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.0625rem;
  -moz-letter-spacing: 0.0625rem;
  -ms-letter-spacing: 0.0625rem;
  letter-spacing: 0.0625rem;
  top: 2.8125rem;
}

.c8 {
  position: absolute;
  width: 1rem;
  height: 1rem;
  background-color: #4475E3;
  border-radius: 50%;
  box-shadow: 0 0 0.5rem rgba(32,87,213,0.25);
  top: -0.5rem;
  left: 0;
  opacity: 1;
}

.c0 {
  margin-top: 1.5rem;
}

<body>
  <div>
    <div
      class="c0"
    >
      <div
        class="c1"
        data-testid="survey-preview-slider"
      >
        <div
          class="c2"
        >
          <div
            class="c3"
            data-testid="survey-preview-slider-point"
          >
            <div
              class="c4"
            />
          </div>
          <div
            class="c5"
            data-testid="survey-preview-slider-point"
          >
            <div
              class="c4"
            />
          </div>
          <div
            class="c5"
            data-testid="survey-preview-slider-point"
          >
            <div
              class="c4"
            />
          </div>
          <div
            class="c5"
            data-testid="survey-preview-slider-point"
          >
            <div
              class="c4"
            />
          </div>
          <div
            class="c5"
            data-testid="survey-preview-slider-point"
          >
            <div
              class="c4"
            />
          </div>
          <div
            class="c5"
            data-testid="survey-preview-slider-point"
          >
            <div
              class="c4"
            />
          </div>
          <div
            class="c5"
            data-testid="survey-preview-slider-point"
          >
            <div
              class="c4"
            />
          </div>
          <div
            class="c5"
            data-testid="survey-preview-slider-point"
          >
            <div
              class="c4"
            />
          </div>
          <div
            class="c5"
            data-testid="survey-preview-slider-point"
          >
            <div
              class="c4"
            />
          </div>
          <div
            class="c5"
            data-testid="survey-preview-slider-point"
          >
            <div
              class="c4"
            />
          </div>
          <div
            class="c6"
            data-testid="survey-preview-slider-point"
          >
            <div
              class="c4"
            />
          </div>
        </div>
        <div
          class="c7"
        />
        <div
          class="c8"
        />
        <div
          class="c9"
        >
          <div
            data-testid="survey-preview-slider-min-value"
          >
            0
          </div>
          <div
            data-testid="survey-preview-slider-max-value"
          >
            10
          </div>
        </div>
        <div
          class="c9 c10"
        >
          <div
            data-testid="survey-preview-slider-min-label"
          >
            Label1
          </div>
          <div
            data-testid="survey-preview-slider-max-label"
          >
            Label2
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
